protos/io_descriptors.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+/* Value represents a single instance of the supported datatypes. 
+ * Naming convention: Follows `numpy.dtype` where appropriate.
+ * `numpy.dtype`: https://numpy.org/doc/stable/reference/arrays.dtypes.html#specifying-and-constructing-data-types */
+message Value{
+	oneof dtype{
+		float f4=1;
+		double f8=2;
+		uint32 u4=3;
+		uint64 u8=4;
+		sint32 i4=5;
+		sint64 i8=6;
+		google.protobuf.Timestamp timestamp_=7;
+		google.protobuf.Duration duration_=8;
+		bool bool_=9;
+		string str_=10;
+		bytes bytes_=11;
+		Value value_=12;
+		Array array_=13;
+		Tuple tuple_=14;
+		// TODO: int32, int64, fixed32, fixed64, sfixed32, sfixed64
+	}
+}
+
+/* Tuple represents a repeated field containing data with same or different datatypes */
+message Tuple{
+	repeated Value value_=1;
+}
+
+// TODO: message complex types
+
+/* Array contains a dtype which identifies the type of the array.
+ * The repeated field for the identified dtype contains the array.
+ * Naming convention: Follows `numpy.dtype` where appropriate.
+ * `numpy.dtype`: https://numpy.org/doc/stable/reference/arrays.dtypes.html#specifying-and-constructing-data-types */
+message Array {
```

Comment:
```suggestion
message NumpyNdarray {
```

Also i think the name should be `protos/io_descriptors/numpy.proto`

Reply:
I think NumpyNdarray provides a better distinction than normal Array. Since a lot of the proto message is Numpy specific, thus the name `NumpyNdarray` makes it more clear than just `Array`.

Reply:
We don't do this in JSON, and I'm not sure we want to do this here. Numpy is what's being used in the server under the hood to process generic array data coming from clients, I don't think clients should care what the input format is.

Reply:
I made the change to `NumpyNdarray`. If you guys want, I can change it back to `Array`